### PR TITLE
Revise docs for Auto-packets and auto-components

### PR DESCRIPTION
**Pull request by chanmosq**
## Change summary
Note: This PR contains broken links, however that page is being added in another PR (https://github.com/o3de/o3de.org/pull/2004) and (https://github.com/o3de/o3de.org/pull/2005)

This PR contains changes that are revived from an old PR (https://github.com/o3de/o3de.org/pull/1365) and incorporates previous feedback, as well as newly added information.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 1]
```diff
@@ -58,10 +58,19 @@ Since each entity with a multiplayer component is replicated across the network,
...
 
-In addition to the roles described above, O3DE has an additional role found only in multiserver instances:
+* **Autonomous** (`NetEntityRole::Autonomous`): A role with the _illusion_ of write access. Autonomous roles are usually assigned to components directly under local user control. These roles receive a larger amount of network information than a Client role. Autonomous roles can also take advantage of predictive networking.
 
-* **ServerProxy**: Provided to servers which *don't* have authority over a particular component. The proxy dummies out the networking properties which the server should treat as if they were in the Client role. In the event of a server crash, proxy servers can fully assume the Server role as a hot backup.
+* **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full write access and sees all network properties on the component.
+In addition to the previously described roles, O3DE has an additional role found only in multiserver instances:
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:66]: Need a blank line here.
```suggestion

In addition to the previously described roles, O3DE has an additional role found only in multiserver instances:
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 2]
```diff
@@ -58,10 +58,19 @@ Since each entity with a multiplayer component is replicated across the network,
...
+
+* **Server** (`NetEntityRole::Server`): Provided to servers that *don't* have authority over a particular component. The proxy dummies out the networking properties, which the server should treat as if they were in the Client role.
+
+## Related topics
+
+| Topic | Description |
+|--|--|
+| [Networking](/docs/user-guide/networking/) | Learn about `AzNetworking`, the core networking framework that O3DE uses, and that the Multiplayer Gem is built on top of. |
+| [Automate Source Generation from Templates with AzAutoGen](/docs/user-guide/engine/autogen/) | Learn about the AzAutoGen code generation system used to create [auto-components](./autocomponents) for projects using the Multiplayer Gem. |
+| [Open 3D Engine Multiplayer Gem API Reference](/docs/api/gems/multiplayer/) | The C++ API reference for classes that the Multiplayer Gem provides. |
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:76]: Consider shortening
```suggestion
| [Multiplayer Gem API Reference](/docs/api/gems/multiplayer/) | The C++ API reference for classes that the O3DE Multiplayer Gem provides. |
```

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/networking/autopackets.md [hunk 3]
```diff
@@ -50,7 +51,7 @@ The `Member` tag defines data on the packet. This is the primary mechanism for d
 
 ### Include
 
-The `Include` tag is used to generate the `#includes` of the C++ code. Use an `Include` tag for each header that your generated classes will use.
+AzAutoGen uses the `Include` tag to generate the `#includes` of the C++ code. Use an `Include` tag for each header that your generated classes will use
```
Comment (sptramer) [inline content/docs/user-guide/networking/autopackets.md:54]:
```suggestion
AzAutoGen uses the `Include` tag to generate the `#includes` of the C++ code. Use an `Include` tag for each header used by the generated classes.
```

**Review by sptramer — CHANGES_REQUESTED** [review]
Minor copy change request, looks good!

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 4]
```diff
@@ -58,10 +58,20 @@ Since each entity with a multiplayer component is replicated across the network,
...
 
-In addition to the roles described above, O3DE has an additional role found only in multiserver instances:
+* **Autonomous** (`NetEntityRole::Autonomous`): A role with the _illusion_ of write access. Autonomous roles are usually assigned to components directly under local user control. These roles receive a larger amount of network information than a Client role. Autonomous roles can also take advantage of predictive networking.
 
-* **ServerProxy**: Provided to servers which *don't* have authority over a particular component. The proxy dummies out the networking properties which the server should treat as if they were in the Client role. In the event of a server crash, proxy servers can fully assume the Server role as a hot backup.
+* **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full write access and sees all network properties on the component.
```
Comment (JwGedit) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:65]: For this statement "It has full write access and sees all network properties on the component" -- does "sees" mean that it detects all network properties on the component?

**Review by JwGedit — CHANGES_REQUESTED** [review]
Edit review complete

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md (hunk 4 above)
Comment (chanmosq) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:65]:
```suggestion
* **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full read and write access to all network properties on the component.
```

**Review by JwGedit — APPROVED** [review]

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 5]
```diff
@@ -58,10 +58,20 @@ Since each entity with a multiplayer component is replicated across the network,
...
-* **Autonomous**: A role with the _illusion_ of write access. Autonomous roles are usually assigned to components directly under local user control, and receive a larger amount of network information than a Client role. Autonomous roles can also take advantage of predictive networking.
-* **Authority**: This role has ultimate authority. It has full write access, and sees all network properties on the component.
+* **Client** (`NetEntityRole::Client`): The lowest privilege role for a component. The smallest possible subset of network properties are replicated to this role, and its behavior is strictly read-only. Use this role when working with entities that should contain only presentation logic and act as a proxy for invoking RPCs. Examples of entities that should use this role are AIs and entities under the control of another host.
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:61]: This may be inaccurate, please check with the networking team.

Comment (kberg-amzn) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:61]: Generally accurate. The only change I would suggest is that you don't get to chose the role a networked entity is assigned, that's all handled automatically by the multiplayer gem itself. Hence I might change it to be more like this:
`* **Client** (`NetEntityRole::Client`): The lowest privilege role for a component. The smallest possible subset of network properties are replicated to this role, and its behavior is strictly read-only. Use this role when working with entities that should contain only presentation logic and act as a proxy for invoking RPCs. Examples of entities that should use this role are AIs and entities under the control of another host.`

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 6]
```diff
@@ -58,10 +58,20 @@ Since each entity with a multiplayer component is replicated across the network,
...
+* **Autonomous** (`NetEntityRole::Autonomous`): A role with the _illusion_ of write access. Autonomous roles are usually assigned to components directly under local user control. These roles receive a larger amount of network information than a Client role. Autonomous roles can also take advantage of predictive networking.
 
-* **ServerProxy**: Provided to servers which *don't* have authority over a particular component. The proxy dummies out the networking properties which the server should treat as if they were in the Client role. In the event of a server crash, proxy servers can fully assume the Server role as a hot backup.
+* **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full read and write access to all network properties on the component.
+
+In addition to the previously described roles, O3DE has an additional role found only in multiserver instances:
+
+* **Server** (`NetEntityRole::Server`): Provided to servers that *don't* have authority over a particular component. The proxy dummies out the networking properties, which the server should treat as if they were in the Client role.
```
Comment (kberg-amzn) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:69]: Provided to servers that *don't* have authority over a particular entity. This role is strictly read-only, and all interaction with these entities should be handled using RPCs.

**Review by JwGedit — APPROVED** [review]

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md [hunk 7]
```diff
@@ -4,9 +4,7 @@ description: A reference for defining Open 3D Engine multiplayer state through a
...
-
-In order to enable auto-component builds for your project, follow the instructions in [Multiplayer Project Configuration](./configuration).
+*Auto-components* provide a convenient way to create multiplayer components that require network synchronization. Using [AzAutoGen](/docs/user-guide/programming/autogen), you can process auto-component files found inside of your project during builds to create C++ classes for components and controllers that provide network replication and remote function calls. Auto-components also manage edit and behavior context bindings so that the bound component shows in **O3DE Editor** and works with **Open 3D Engine (O3DE)** scripting.
```
Comment (AMZN-Gene) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md:7]: hyperlink edit context? https://www.o3de.org/docs/user-guide/programming/components/reflection/edit-context/
hyperlink behavior context? https://www.o3de.org/docs/user-guide/programming/components/reflection/behavior-context/

**Review by AMZN-Gene — APPROVED** [review]

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md (hunk 5 above)
Comment (chanmosq) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md]: If that's the case, I'd go even further to specify that the Multiplayer Gem automatically assigns and handles the roles of networked entities at the top of this section. 

@kberg-amzn You omitted the example sentence in your suggested rewrite. I just wanted to verify whether that example still sounds accurate?

**Review by sptramer — APPROVED** [review]

**Review by willihay — APPROVED** [review]

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 8]
```diff
@@ -15,53 +15,63 @@ Some of the features offered in the O3DE Multiplayer Gem include:
...
 {{< /important >}}
 
 ## Network properties
 
-The O3DE Multiplayer Gem uses *delta replication* to transmit property changes across hosts. The delta replication method uses a system where a server maintains information about both each client's last acknowledged state and the current world state, then sends the differences between the two across the network. When the client receives this information, it updates the world state if the packet was received in the correct sequence.
+The O3DE Multiplayer Gem uses *delta replication* to transmit property changes across hosts. The delta replication method uses a system where a server maintains information about both the last acknowledged state of each client and the current world state, and then sends the differences between the two states across the network. When the client receives this information, it updates the world state if the packet was received in the correct sequence.
 
 One important consideration of this style of replication is that it requires the client to send an acknowledgement (`ACK`) message back to the server. The O3DE AzNetworking framework offers reliable UDP packets, including acknowledgement.
```
Comment (chanmosq) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:33]:
```suggestion
One important consideration of this style of replication is that it requires the client to send an acknowledgement (`ACK`) message back to the server. The O3DE AzNetworking framework offers reliable UDP packets, including acknowledgement. ACK vector replication allows the AzNetworking layer to be fully aware of every packet that was received or dropped. 
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md [hunk 9]
```diff
@@ -15,53 +15,63 @@ Some of the features offered in the O3DE Multiplayer Gem include:
...
-1. If the world state is different from the last state which resulted in an `ACK`, generate a delta for that client.
+1. If the world state is different from the last state that resulted in an `ACK`, generate a delta for that client.
 1. Transmit the delta to the client.
 
-Clients would then update themselves from a delta by:
+Then, clients update themselves from a delta by:
 
 1. Checking if the received delta is in sequence.
 1. Extracting the delta information and updating the world state.
 1. Sending an `ACK` to the server indicating the world was updated.
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:45]:
```suggestion
1. Sending an `ACK` to the server, indicating to the AzNetworking layer that the world was updated.
```

[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md:45]: I made a suggestion above to clarify this part, see line 33.

**Review by kberg-amzn — APPROVED** [review]

**Review by micronAMZN — APPROVED** [review]

**Pull request merged**